include/mxnet/resource.h

```diff
@@ -61,12 +61,12 @@ struct Resource {
    * \return the mshadow random number generator requested.
    * \tparam xpu the device type of random number generator.
    */
-  template<typename xpu>
-  inline mshadow::Random<xpu>* get_random(
+  template<typename xpu, typename DType>
```

Comment:
I will assume we only support random with DType float for now, cast can always be applied